merkle-maker/merkle_maker/main.py

```diff
@@ -0,0 +1,206 @@
+
+from merkle_maker.redis import RedisManager
+
+logger = logging.getLogger(__name__)
+
+TIME_TO_WAIT_BETWEEN_BLOCK_NUMBER_POLLING = 1
+
+
+async def main(
+    network: Literal["mainnet", "sepolia"],
+    redis_host: str,
+    publisher_name: str,
+    publisher_address: str,
+    private_key: str | Tuple[str, str],
+    block_interval: int,
+    rpc_url: Optional[HttpUrl] = None,
+) -> None:
+    logger.info("🔨 Setting up clients and Redis connection...")
+    pragma_client = PragmaOnChainClient(
+        chain_name=network,
+        network=network if rpc_url is None else rpc_url,
+        account_contract_address=publisher_address,
+        account_private_key=private_key,
+    )
+
+    redis_host, redis_port = redis_host.split(":")
+    redis_manager = RedisManager(host=redis_host, port=redis_port)
+
+    fetcher_client = FetcherClient()
+    deribit_fetcher = DeribitOptionsFetcher(
+        pairs=[
+            Pair.from_tickers("BTC", "USD"),  # type: ignore[list-item]
+            Pair.from_tickers("ETH", "USD"),  # type: ignore[list-item]
+        ],
+        publisher=publisher_name,
+    )
+    fetcher_client.add_fetcher(deribit_fetcher)
+
+    logger.info("🧩 Starting the Merkle Maker...")
+    await _publish_merkle_feeds_forever(
+        pragma_client=pragma_client,
+        fetcher_client=fetcher_client,
+        redis_manager=redis_manager,
+        block_interval=block_interval,
+    )
+
+
+async def _publish_merkle_feeds_forever(
+    pragma_client: PragmaOnChainClient,
+    fetcher_client: FetcherClient,
+    redis_manager: RedisManager,
+    block_interval: int,
+) -> Never:
+    """
+    Publish a new Merkle Feed on chain every [block_interval] block(s) forever.
+    We store the merkle tree and the options used to generate the merkle root
+    to a Redis database that will get consumed by our Rust service.
+    """
+    deribit_fetcher: DeribitOptionsFetcher = fetcher_client.fetchers[0]  # type: ignore[assignment]
+    while True:
+        # TODO: In case of a restart, check if a merkle feed already exists for the
+        # current block.
+        current_block = await pragma_client.get_block_number()
+        logger.info(f"Current block: {current_block}")
+
+        logger.info("🔍 Fetching the deribit options...")
+        entries = await fetcher_client.fetch()
+
+        logger.info("🎣 Publishing the merkle root onchain...")
+        try:
+            await pragma_client.publish_entries(entries)  # type: ignore[arg-type]
+            logger.info("... done!")
+        except Exception:
+            # TODO: remove this part when the contract has been updated
+            logger.warning("Could not publish! Contract not yet updated.")
+
+        # TODO: move this block in another thread so we loose 0 time on this?
+        # NOTE: it is VERY fast though so probably not needed
+        logger.info("🏭 Storing the merkle tree & options in Redis...")
+        redis_manager.store_latest_data(deribit_fetcher.get_latest_data())
```

Comment:
checked documentations/online and looks like the only way the `SET` could fail is because the connection to the Redis instance is broken or the Redis instance is out of memory. For both cases we should raise an Error & stop the app imo.

Added that